src/main/java/net/openhft/posix/internal/jnr/JNRPosixAPI.java

```diff
@@ -19,6 +21,14 @@
     static final String STANDARD_C_LIBRARY_NAME = NATIVE_PLATFORM.getStandardCLibraryName();
     static final Pointer NULL = Pointer.wrap(RUNTIME, 0);
 
+    static final int SYS_mlock2; // mlock2 syscall value
+    static {
+        // These cover the main cases. Full list under https://github.com/torvalds/linux/tree/master/arch
+        SYS_mlock2 = Jvm.isArm() ? 390
```

Comment:
I think the primary target should be ARM 64-bit in which case it should be 284. Our ARM agent is `armhf` which indicates 64-bit I presume. Other numbers seem correct. I also wonder what's the dimension of first arg for 32-bit targets - do we have a test that would fail in absence of locking?

Reply:
Maybe armhf is not 64-bit after all...

Reply:
The numbers coded in the static block for the various platforms were derived from the linux source (dev12:/dev/shm/roger/linux).All (LInux) ARM variants are 390 for mlock2. macOS doesn't support mlock2 so we degrade to mlock there (ie the SYS lookup is not relevant for macOS M1).

Perhaps you could cross check these. In particular:
```
./x86/entry/syscalls/syscall_64.tbl:325 common  mlock2                  sys_mlock2
./x86/entry/syscalls/syscall_32.tbl:376 i386    mlock2                  sys_mlock2
./arm64/include/asm/unistd32.h:#define __NR_mlock2 390
./arm/tools/syscall.tbl:390     common  mlock2                  sys_mlock2
```